### ISSUE TITLE
[FR] Integration with Frontmatter Title Plugin

### ISSUE THREAD

**Issue by Moyf** [oxdc/obsidian-vertical-tabs#27]
Hello! Could you integrate this plugin with https://github.com/snezhig/obsidian-front-matter-title so that we can see the `title` property rather than the filename?
This plugin provides an API : https://github.com/snezhig/front-matter-plugin-api-provider

**Comment by oxdc** [comment]
Good idea! I'm afraid that integrations with other plugins, however, might have to be postponed for a little while. I'll look into it after fixing all known issues and getting the core functionality done. Please be patient.

**Comment by Moyf** [comment]
> Good idea! I'm afraid that integrations with other plugins, however, might have to be postponed for a little while. I'll look into it after fixing all known issues and getting the core functionality done. Please be patient.

Of course! No worry, take your time ; )

**Comment by attilaszkrs** [comment]
This feature request started on the wrong foot. The issue is not integrating Vertical Tabs with another plugin; it's about displaying a note's frontmatter title. This functionality relies on standard YAML frontmatter, a core feature of Obsidian, and doesn't require a third-party plugin to access. This feature would greatly enhance Vertical Tabs for users who decoupled or 'internalized' titles" from filenames. Title should be: "option to display front matter title instead of file name"

**Comment by oxdc** [comment]
Hi @attilaszkrs ! May I ask which solution do you use or prefer? With or without the frontmatter title plugin?

The advantage of Frontmatter Title is that it comes with more features such as replacing titles in the file browser. Otherwise Vertical Tabs has to reimplement some features.

**Comment by Moyf** [comment]
For me both solutions work, since the result is "show title instead of filename".

Of course, I understand that some users don't want to rely on additional plugins -
First, however, as @oxdc said, the **Frontmatter title** plugin implements some features in its own right, and integrating with it avoids the need to reinvent the wheel.

Examples include customizable `title` frontmatter property name, fallback frontmatter name, and more complex function handling:
<img width="1065" alt="image" src="https://github.com/user-attachments/assets/43f59ce7-2a84-40a7-87e3-75cada256b80">

For example, relying on the `Function V2` Processor, I can display the name of "Today's DailyNote" as `⬇️ Today`:
<img width="457" alt="image" src="https://github.com/user-attachments/assets/da5d1d08-6dca-4047-b371-b5476df36c0e">
This is easier to identify. And because it's dynamic, it can not be achieved by a fixed `title`.

---

Secondly, I personally think that the need to "display `title`" should be unified.
If a person wants to display the title in **vertical tabs**, then it is likely that he will also want to display the title in other places - tabs, title bars (etc.), so this is to "display title" as a special problem, with a separate solution to unify - **Vertical tabs** are only compatible with this solution.
<img width="807" alt="image" src="https://github.com/user-attachments/assets/3eab75c9-60f6-419d-9687-fc1a7d2cc95f">

Still, both solutions are acceptable for me! It depends on what @oxdc thinks 😊
